addons/pos_viva_wallet/controllers/main.py

```diff
@@ -3,16 +3,22 @@
 import json
 from odoo import http, _
 from odoo.http import request
+from odoo.tools import consteq
 
 _logger = logging.getLogger(__name__)
 
 
 class PosVivaWalletController(http.Controller):
-    @http.route('/pos_viva_wallet/notification', type='http', auth='none', csrf=False)
+    @http.route('/pos_viva_wallet/notification', type='http', auth='none', csrf=False, readonly=False)
     def notification(self, company_id, token):
         _logger.info('notification received from Viva Wallet')
 
-        payment_method_sudo = request.env['pos.payment.method'].sudo().search([('use_payment_terminal', '=', 'viva_wallet'), ('company_id.id', '=', company_id), ('viva_wallet_webhook_verification_key', '=', token)], limit=1)
+        payment_method_sudo = None
+        viva_payment_methods = request.env['pos.payment.method'].sudo().search([('use_payment_terminal', '=', 'viva_wallet'), ('company_id.id', '=', company_id)])
+        for payment_method in viva_payment_methods:
+            if consteq(payment_method.viva_wallet_webhook_verification_key, token):
+                payment_method_sudo = payment_method
+                break
```

Comment:
```suggestion
        payment_method_sudo = next((pm for pm in viva_payment_methods if consteq(pm.viva_wallet_webhook_verification_key, token)), None)
```

Would avoid `for/break`